File: suc/kotlin/057.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "086_SleepCycle_en"
private const val UI_TYPE = "Health Auth"
private const val STYLE_THEME = "Modern Blue"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF2563EB)
        val secondary = Color(0xFF38BDF8)
        val tertiary = Color(0xFF60A5FA)
        val background = Color(0xFFF8FAFC)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFE2E8F0)
        val outline = Color(0xFFCBD5E1)
        val success = Color(0xFF22C55E)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFF0F172A)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF0F172A)
        val onSurface = Color(0xFF1E293B)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.Bold)
        val headline = TextStyle(fontSize = 20.sp, fontWeight = FontWeight.SemiBold)
        val title = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(8.dp)
        val medium = RoundedCornerShape(12.dp)
        val large = RoundedCornerShape(16.dp)
    }
    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
        val xxl = 36.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(2.dp, 4.dp, 2.dp, 0.12f)
        val level2 = ShadowSpec(4.dp, 8.dp, 4.dp, 0.14f)
        val level3 = ShadowSpec(8.dp, 12.dp, 6.dp, 0.16f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    Scaffold(
        contentWindowInsets = androidx.compose.foundation.layout.WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Text("Sleep Cycle", style = MaterialTheme.typography.displayLarge, color = AppTokens.Colors.onBackground)
                },
                navigationIcon = {
                    IconButton(onClick = {}) {
                        Box(
                            modifier = Modifier
                                .size(24.dp)
                                .background(AppTokens.Colors.primary, CircleShape),
                            contentAlignment = Alignment.Center
                        ) {
                            Text("<", color = AppTokens.Colors.onPrimary, style = MaterialTheme.typography.labelMedium)
                        }
                    }
                },
                actions = {
                    IconButton(onClick = {}) {
                        Box(
                            modifier = Modifier
                                .size(24.dp)
                                .background(AppTokens.Colors.secondary, CircleShape),
                            contentAlignment = Alignment.Center
                        ) {
                            Text("?", color = AppTokens.Colors.onSecondary, style = MaterialTheme.typography.labelMedium)
                        }
                    }
                }
            )
        },
        containerColor = AppTokens.Colors.background
    ) { padding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .background(
                    Brush.verticalGradient(
                        listOf(AppTokens.Colors.secondary.copy(alpha = 0.15f), AppTokens.Colors.primary.copy(alpha = 0.1f))
                    )
                )
                .padding(AppTokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xl),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Box(
                modifier = Modifier
                    .size(160.dp)
                    .background(AppTokens.Colors.surface, CircleShape)
                    .border(2.dp, AppTokens.Colors.outline, CircleShape),
                contentAlignment = Alignment.Center
            ) {
                Text("😴", fontSize = 48.sp)
            }
            Text("Welcome Back", style = MaterialTheme.typography.headlineMedium, color = AppTokens.Colors.onSurface)
            Text("Track your sleep patterns and improve rest quality.", style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.onSurface)
            Button(
                onClick = {},
                modifier = Modifier.fillMaxWidth(0.8f).height(52.dp),
                shape = AppTokens.Shapes.large,
                colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary, contentColor = AppTokens.Colors.onPrimary)
            ) {
                Text("Sign In", style = MaterialTheme.typography.titleMedium)
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFF8FAFC)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/057.swift
import SwiftUI

// MARK: - AppTokens
// This struct defines a consistent set of design tokens for colors, typography, shapes, and spacing,
// mirroring the AppTokens object in the Kotlin version for atomic correspondence.
struct AppTokens {
    struct Colors {
        static let primary = Color(red: 37/255, green: 99/255, blue: 235/255) // #2563EB
        static let secondary = Color(red: 56/255, green: 189/255, blue: 248/255) // #38BDF8
        static let tertiary = Color(red: 96/255, green: 165/255, blue: 250/255) // #60A5FA
        // 加深背景蓝色
        static let background = Color(red: 230/255, green: 240/255, blue: 255/255) // #E6F0FF - 更深的蓝色背景
        static let surface = Color(red: 255/255, green: 255/255, blue: 255/255) // #FFFFFF
        static let surfaceVariant = Color(red: 226/255, green: 232/255, blue: 240/255) // #E2E8F0
        static let outline = Color(red: 203/255, green: 213/255, blue: 225/255) // #CBD5E1
        static let success = Color(red: 34/255, green: 197/255, blue: 94/255) // #22C55E
        static let warning = Color(red: 245/255, green: 158/255, blue: 11/255) // #F59E0B
        static let error = Color(red: 239/255, green: 68/255, blue: 68/255) // #EF4444
        static let onPrimary = Color(red: 255/255, green: 255/255, blue: 255/255) // #FFFFFF
        static let onSecondary = Color(red: 15/255, green: 23/255, blue: 42/255) // #0F172A
        static let onTertiary = Color(red: 255/255, green: 255/255, blue: 255/255) // #FFFFFF
        static let onBackground = Color(red: 15/255, green: 23/255, blue: 42/255) // #0F172A
        static let onSurface = Color(red: 30/255, green: 41/255, blue: 59/255) // #1E293B
    }

    struct TypographyTokens {
        // Using .system font with specified size and weight to match TextStyle in Compose.
        static let display = Font.system(size: 28, weight: .bold)
        static let headline = Font.system(size: 20, weight: .semibold)
        static let title = Font.system(size: 16, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular)
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        // Corner radii are represented as CGFloat values.
        static let small: CGFloat = 8
        static let medium: CGFloat = 12
        static let large: CGFloat = 16
    }

    struct Spacing {
        // Spacing values are represented as CGFloat values.
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 12
        static let lg: CGFloat = 16
        static let xl: CGFloat = 24
        static let xxl: CGFloat = 36
    }
}

// MARK: - RootScreen
// This view translates the RootScreen Composable from Kotlin to SwiftUI.
// It uses GeometryReader to accurately calculate dynamic sizes like button width.
struct RootScreen: View {
    var body: some View {
        GeometryReader { geometry in // Used for dynamic width calculations to match `fillMaxWidth`
            VStack(spacing: 0) { // Main vertical stack to arrange top bar and content
                // Top Bar (equivalent to Compose's CenterAlignedTopAppBar)
                HStack {
                    // Navigation Icon (Left)
                    Button(action: {}) {
                        ZStack {
                            Circle()
                                .fill(AppTokens.Colors.primary)
                                .frame(width: 24, height: 24) // Fixed size for the circular background
                            Text("<")
                                .foregroundColor(AppTokens.Colors.onPrimary)
                                .font(AppTokens.TypographyTokens.label)
                        }
                    }
                    .padding(.leading, AppTokens.Spacing.lg) // Padding to match the Android version

                    Spacer() // Pushes title to center and icons to edges

                    // Title
                    Text("Sleep Cycle")
                        .font(AppTokens.TypographyTokens.display)
                        .foregroundColor(AppTokens.Colors.onBackground)

                    Spacer() // Pushes title to center and icons to edges

                    // Actions Icon (Right)
                    Button(action: {}) {
                        ZStack {
                            Circle()
                                .fill(AppTokens.Colors.secondary)
                                .frame(width: 24, height: 24) // Fixed size for the circular background
                            Text("?")
                                .foregroundColor(AppTokens.Colors.onSecondary)
                                .font(AppTokens.TypographyTokens.label)
                        }
                    }
                    .padding(.trailing, AppTokens.Spacing.lg) // Padding to match the Android version
                }
                .frame(height: 56) // Standard height for Material3 TopAppBar
                .background(AppTokens.Colors.surface) // 顶部栏使用纯白色背景
                // End Top Bar

                // Main Content Area (equivalent to Scaffold content Column)
                VStack(spacing: AppTokens.Spacing.xl) { // Vertical arrangement with spacedBy
                    Spacer() // Pushes content to center vertically

                    // Central Emoji Box
                    ZStack {
                        Circle()
                            .fill(AppTokens.Colors.surface)
                            .frame(width: 160, height: 160) // Fixed size
                            .overlay(
                                Circle()
                                    .stroke(AppTokens.Colors.outline, lineWidth: 2) // Border
                            )
                        Text("😴")
                            .font(.system(size: 48)) // Fixed font size for emoji
                    }

                    // Welcome Back Text
                    Text("Welcome Back")
                        .font(AppTokens.TypographyTokens.headline)
                        .foregroundColor(AppTokens.Colors.onSurface)

                    // Description Text
                    Text("Track your sleep patterns and improve rest quality.")
                        .font(AppTokens.TypographyTokens.body)
                        .foregroundColor(AppTokens.Colors.onSurface)
                        .multilineTextAlignment(.center) // Center-align multi-line text

                    // Sign In Button
                    Button(action: {}) {
                        Text("Sign In")
                            .font(AppTokens.TypographyTokens.title)
                            .foregroundColor(AppTokens.Colors.onPrimary)
                            .frame(maxWidth: .infinity) // Text fills the button's width
                            .frame(height: 52) // Fixed height for the button content
                            .background(AppTokens.Colors.primary) // Button background color
                            .cornerRadius(AppTokens.Shapes.large) // Rounded corners
                    }
                    // Calculate button width to atomically match Kotlin's `fillMaxWidth(0.8f)`
                    // The Kotlin Column has `padding(AppTokens.Spacing.lg)`.
                    // The button then takes 80% of the width *inside* that padding.
                    .frame(width: (geometry.size.width - 2 * AppTokens.Spacing.lg) * 0.8)

                    Spacer() // Pushes content to center vertically
                }
                .padding(.horizontal, AppTokens.Spacing.lg) // Column's horizontal padding
                .frame(maxWidth: .infinity, maxHeight: .infinity) // Fill remaining space in the VStack
                .background(
                    // 加深渐变背景
                    LinearGradient(
                        gradient: Gradient(colors: [
                            AppTokens.Colors.background, // 使用更深的背景色
                            AppTokens.Colors.background // 单色背景，更深的蓝色
                        ]),
                        startPoint: .top,
                        endPoint: .bottom
                    )
                )
            }
            .background(AppTokens.Colors.background) // 整体背景使用更深的蓝色
            .edgesIgnoringSafeArea(.all) // Ensures the content extends to the very edges of the screen
        }
    }
}

// MARK: - App Entry Point
// This is the main entry point for the SwiftUI application.
// It sets up the RootScreen and applies global modifiers for full-screen display and status bar visibility.
@main
struct SleepCycleApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
                // Hide the status bar to achieve a full-screen experience, compatible with iOS 16.0.
                .statusBarHidden(true)
        }
    }
}